core/src/oauth/connection.rs

```diff
@@ -144,11 +144,16 @@ pub trait Provider {
     async fn finalize(
         &self,
         connection: &Connection,
+        credentials: Option<Credential>,
```

Comment:
Should we call that `related_credentials` everywhere? 
I think this could make it easier to understand what it's in here for, since we have to change the signature for all providers and we use it only for Salesforce. WDYT?